kingpin/bin/deploy.py

```diff
@@ -71,25 +75,34 @@ def main():
 
     env_tokens = dict(os.environ)
 
-    try:
-        json_file = options.json or sys.argv[1] if sys.argv else None
-    except Exception as e:
-        kingpin_fail(
-            '%s You must specify --json or provide it as first argument.' % e)
+    # Sanity check - did the user supply both a JSON script && an individual
+    # actor? If so, print the help!
+    if args.json and args.actor:
+        kingpin_fail('You may only specify --actor or --json, not both!')
 
+    if args.actor:
+        json_file = utils.get_script_from_args(args)
```

Comment:
I'm not sure I like this approach. `arg.actor` is telling you which actor to use, but instead of using it you create a Macro that uses the actor. I recommend rewriting this part to overwrite `dry_actor` and `runner` with the actual actor that is being executed.